frontend/src/components/ExtractUploadFile.tsx

```diff
@@ -23,23 +35,11 @@ export const ExtractUploadFile: React.FC<ExtractUploadFileProps> = ({
   onUploadComplete,
 }) => {
   const id = useId();
-  const { addFile } = useFiles();
-  const [template, setTemplate] = useState<string>("");
-  const [uploadProgress, setUploadProgress] = useState<number>(0);
+  const { addFile, clearFiles, files, setSelectedTemplates, selectedTemplates } = useFiles();
+  const navigate = useNavigate();
   const [templates, setTemplates] = useState<Template[]>([]);
-  const [uploadedFile, setUploadedFile] = useState<File | null>(null);
-
-  // Load templates from local storage, and if none are found, load test data
-  const loadTemplatesFromLocalStorage = () => {
-    const storedTemplates = localStorage.getItem("templates");
-    if (storedTemplates) {
-      const parsedTemplates = JSON.parse(storedTemplates);
-      setTemplates(parsedTemplates);
-    } else {
-      loadTemplatesTestData();
-    }
-  };
-
+  const [isUploadComplete, setIsUploadComplete] = useState<boolean>(false);
```

Comment:
Might need a better state solution soon like contextAPI. This is ok for now.

Reply:
sounds good